components/speedreader/rust/lib/src/readability/src/extractor.rs

```diff
@@ -203,6 +416,137 @@ pub fn clean_title(dom: &Sink, title: String) -> String {
     }
 }
 
+#[derive(Error, Debug, PartialEq)]
+pub enum JsonLdError {
+    #[error("Could not parse json-ld: `{0}`")]
+    ParseError(String),
+    #[error("Missing @context field")]
+    MissingContext,
+    #[error("Invalid @context: Does not match http://schema.org")]
+    InvalidContext,
+    #[error("Missing @type field")]
+    MissingType,
+    #[error("Invalid @type")]
+    InvalidType,
+}
+
+impl From<serde_json::Error> for JsonLdError {
+    fn from(err: serde_json::Error) -> Self {
+        JsonLdError::ParseError(err.to_string())
+    }
+}
+
+#[inline(never)]
+fn try_parse_author(v: &serde_json::Value) -> Option<String> {
+    match v {
+        serde_json::Value::String(s) => {
+            if s.starts_with("[{") {
+                // Looks like it's nested JSON. I have no idea why sites do this. It's so stupid, but some do...
+                let p = serde_json::from_str(s).ok()?;
+                try_parse_author(&p)
+            } else {
+                Some(s.to_string())
+            }
+        }
+        serde_json::Value::Array(a) => {
+            if !a.is_empty() {
+                Some(
+                    a.iter()
+                        .filter_map(|e| try_parse_author(e))
+                        .collect::<Vec<_>>()
+                        .join(", "),
```

Comment:
Suggestion, instead of formatting the list of authors like this (strings with comma separators), can you just output as a structured, HTML list, and then do layout / display / formatting in CSS?  Partially it'll make formatting more flexible, but partially it'll disambiguate the cases where an author has a comma in their displayed name:

E.g. <li>Snyder, Peter</li><li>Smith, Mark E</li> vs "Snyder, Peter, Smith, Mark E" etc

Reply:
interesting thought. this isn't in the design, so we can go over this during Wednesday meeting